src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1580,138 +1625,90 @@ private String getTokenHash(String token) {
 
-            // Temporarily disable the family client id feature. 
-            final boolean isFamilyClientIdFeatureEnabled = false;
-            if (isFamilyClientIdFeatureEnabled) {
-                // If still cannot find an item, try with family clientid
-                // Disable this feature for ADFS authority
-                if (item == null || StringExtensions.IsNullOrBlank(item.getRefreshToken())) {
-                    if (!UrlExtensions.isADFSAuthority(StringExtensions.getUrl(mAuthority))) {
-                        Logger.v(TAG + methodName, "No refresh token found, trying to find family client id item.");
-                        item = findFamilyItemForUser(request.getUserIdentifierType(), userId);
-                    } else {
-                        Logger.v(TAG + methodName, "No refresh token found, skip the family client id item lookup for ADFS authority.");
+                    Logger.v(TAG + methodName, mrrtLookupMsg);
+                    refreshItem = getRefreshTokenFromCache(KeyEntryType.FAMILY_REFRESH_TOKEN_ENTRY, request);
+                    
+                    if (refreshItem == null) {
+                        Logger.v(TAG + methodName, "FRT does not exist, fall back to MRRT if exists.");
+                        refreshItem = mrrtItem;
                     }
                 }
             }
-
-            if (item != null && !StringExtensions.IsNullOrBlank(item.getRefreshToken())) {
-                String refreshTokenHash = getTokenHash(item.getRefreshToken());
-
-                Logger.v(TAG, "Refresh token is available and id:" + refreshTokenHash
-                        + " Key used:" + keyUsed);
-                refreshItem = new RefreshItem(keyUsed, request, item);
-            }
         }
 
         return refreshItem;
     }
-
-    private TokenCacheItem findFamilyItemForUser(final UserIdentifierType userIdentifierType, final String userId) {
-        final String methodName = ":findFamilyItemForUser";
-            
-        if (StringExtensions.IsNullOrBlank(userId)) {
-            Logger.v(TAG + methodName, "User id is not provided, cannot continue with finding family item.");
-            return null;
-        }
-
-        Iterator<TokenCacheItem> allItems = mTokenCacheStore.getAll();
-        if (allItems == null || !allItems.hasNext()) {
-            Logger.v(TAG + methodName, "No items in the cache, cannot continue with finding family item.");
+    
+    /**
+     * Get {@link RefreshItem} based on the given key entry type. 
+     */
+    private RefreshItem getRefreshTokenFromCache(final KeyEntryType keyEntryType, 
+            final AuthenticationRequest authRequest) {
+        final String user = getUserFromAuthenticationRequest(authRequest);
+        if (StringExtensions.IsNullOrBlank(user) && KeyEntryType.FAMILY_REFRESH_TOKEN_ENTRY == keyEntryType) {
+            // Fmaily token should be retrieved specifically by user. If no user is passed in the request, 
```

Comment:
For non-silent request, there may not be any upn passed in, we should still try our best to find something in the token cache, and try it, if it doesn't work, launch the UI then.